aspnetcore/migration/webapi.md

```diff
@@ -4,47 +4,164 @@ author: ardalis
+
+## Create the destination project
+
+Create a new blank solution in Visual Studio and add the ASP.NET 4.x Web API project to migrate:
+
+1. From the **File** menu, select **New** > **Project**.
+1. Select the **Blank Solution** template and select **Next**.
+1. Name the solution *WebAPIMigration*. Select **Create**.
+1. Add the existing *ProductsApp* project to the solution.
+
+Add a new API project to migrate to:
+
+1. Add a new **ASP.NET Core Web Application** project to the solution.
+1. In the **Configure your new project** dialog, Name the project *ProductsCore*, and select **Create**.
+1. In the **Create a new ASP.NET Core Web Application** dialog, confirm that **.NET Core** and **ASP.NET Core 3.1** are selected. Select the **API** project template, and select **Create**.
+1. Remove the *WeatherForecast.cs* and *Controllers/WeatherForecastController.cs* example files from the new *ProductsCore* project.
+
+The solution now contains two projects. The following sections explain migrating the *ProductsApp* project's contents to the *ProductsCore* project.
+
+## Migrate configuration
+
+ASP.NET Core doesn't use the *App_Start* folder or the *Global.asax* file. Additionally, the *web.config* file is added at publish time.
+
+The `Startup` class:
+
+* Replaces *Global.asax*.
+* Handles all app startup tasks.
+
+For more information, see <xref:fundamentals/startup>.
+
+## Migrate models and controllers
+
+The following code shows the `ProductsController` update for ASP.NET Core:
+
+[!code-csharp[](webapi/sample/3.x/ProductsApp/Controllers/ProductsController.cs)]
 
-The *ProductsApp* project includes one controller. The controller inherits from `ApiController` and contains two actions:
+Update the `ProductsController` for ASP.NET Core:
 
-[!code-csharp[](webapi/sample/ProductsApp/Controllers/ProductsController.cs?highlight=28,33)]
+1. Copy *Controllers/ProductsController.cs* and the *Models* folder from the original project to the new one.
+1. Change the copied files' root namespace to `ProductsCore`.
+1. Update the `using ProductsApp.Models;` statement to `using ProductsCore.Models;`.
 
-The `Product` model used by `ProductsController` is a simple class:
+The following components don't exist in ASP.NET Core:
 
-[!code-csharp[](webapi/sample/ProductsApp/Models/Product.cs)]
+* `ApiController` class
+* `System.Web.Http` namespace
+* `IHttpActionResult` interface
+
+Make the following changes:
+
+1. Change `ApiController` to <xref:Microsoft.AspNetCore.Mvc.ControllerBase>. Add `using Microsoft.AspNetCore.Mvc;` to resolve the `ControllerBase` reference.
+1. Delete `using System.Web.Http;`.
+1. Change the `GetProduct` action's return type from `IHttpActionResult` to `ActionResult<Product>`.
+
+Simplify the `GetProduct` action's `return` statement to the following:
+
+```csharp
+return product;
+```
+
+## Configure routing
+
+All ASP.NET Core project templates include attribute routing in the generated code.
```

Comment:
I suggest future-proofing this statement by talking specifically about the API project template. The project templates change quite often.

```suggestion
The ASP.NET Core *API* project template includes endpoint routing configuration in the generated code.
```